File: backend/src/main/java/com/ll/hotel/domain/hotel/room/dto/GetRoomResponse.java
package com.ll.hotel.domain.hotel.room.dto;

import com.ll.hotel.domain.hotel.room.type.BedTypeNumber;
import jakarta.validation.constraints.NotBlank;
import lombok.NonNull;

public record GetRoomResponse(
        long roomId,

        @NotBlank
        String roomName,

        @NonNull
        Integer basePrice,

        @NonNull
        Integer standardNumber,

        @NonNull
        Integer maxNumber,

        @NotBlank
        BedTypeNumber bedTypeNumber,

        @NonNull
        String thumbnailUrl,

        @NonNull
        Integer roomNumber
) {
    public GetRoomResponse(RoomWithImageDto roomWithImageDto) {
        this(
                roomWithImageDto.room().getId(),
                roomWithImageDto.room().getRoomName(),
                roomWithImageDto.room().getBasePrice(),
                roomWithImageDto.room().getStandardNumber(),
                roomWithImageDto.room().getMaxNumber(),
                roomWithImageDto.room().getBedTypeNumber(),
                roomWithImageDto.image() == null
                        ? "/images/default.jpg"
                        : roomWithImageDto.image().getImageUrl(),
                roomWithImageDto.room().getRoomNumber()
        );
    }
}

File: backend/src/main/java/com/ll/hotel/domain/hotel/room/dto/GetRoomResponse.kt
package com.ll.hotel.domain.hotel.room.dto

import com.ll.hotel.domain.hotel.room.type.BedTypeNumber
import jakarta.validation.constraints.NotBlank

data class GetRoomResponse(
    val roomId: Long,

    @field:NotBlank
    val roomName: String,

    val basePrice: Int,

    val standardNumber: Int,

    val maxNumber: Int,

    @field:NotBlank
    val bedTypeNumber: BedTypeNumber,

    val thumbnailUrl: String,

    val roomNumber: Int
) {
    constructor(roomWithImageDto: RoomWithImageDto) : this(
        roomWithImageDto.room.id,
        roomWithImageDto.room.roomName,
        roomWithImageDto.room.basePrice,
        roomWithImageDto.room.standardNumber,
        roomWithImageDto.room.maxNumber,
        roomWithImageDto.room.bedTypeNumber,
        roomWithImageDto.image?.imageUrl ?: "/images/default.jpg",
        roomWithImageDto.room.roomNumber
    )
}